mcs/class/corlib/ReferenceSources/CalendarData.cs

```diff
@@ -43,7 +43,17 @@ public static int nativeGetTwoDigitYearMax (int calID)
 		static bool nativeGetCalendarData (CalendarData data, string localeName, int calendarId)
 		{
 			// TODO: Convert calendar-id to mono runtime calendar-id when it's used
-			return data.fill_calendar_data (localeName.ToLowerInvariant (), calendarId);
+			if (data.fill_calendar_data (localeName.ToLowerInvariant (), calendarId)) {
+				if ((CalendarId)calendarId == CalendarId.HEBREW) {
+					// HACK: Generate leap month names because the native code does not do it
+					// and tests would fail to round-trip dates on leap years.
+					// https://github.com/mono/mono/issues/16623
+					data.saMonthNames = new[] { "Nisan", "Iyar", "Sivan", "Tammuz", "Av", "Elul", "Tishrei", "Cheshvan", "Kislev", "Tevet", "Shevat", "Adar", "" };
+					data.saLeapYearMonthNames = new[] { "Nisan", "Iyar", "Sivan", "Tammuz", "Av", "Elul", "Tishrei", "Cheshvan", "Kislev", "Tevet", "Shevat", "Adar I", "Adar II" };
```

Comment:
The data may be in incorrect order (Biblical vs. Civil). I did some experiments but I have no idea how we end up in the month that it tries to locate. Also, NetFX seems to allow only combination of the Hebrew calendar with the `he-IL` locale so I should change the names to the Hebrew ones.

Reply:
The order is indeed incorrect. We should be in  `Elul` now which is the 13th month on a leap year. Will fix it.

Reply:
Fixed. Turned out that I had one more bug in there.

Apparently `MonthNames` always return 13 months and it's essential for the date/time parsing code. `LeapMonthNames` also has 13 months but the names are slightly different. The regular names are "Adar" and "Adar II", the leap names are "Adar I" and "Adar II".